src/Benchmarks/InvokeBenchmarks/MarshalBenchmark.cs

```diff
@@ -0,0 +1,164 @@
+                    return true;
+                }
+                
+                if (proc == "Test2")
+                {
+                    addr = (nint)(delegate* unmanaged[Cdecl]<int>)(&WithReturn);
+                    return true;
+                }
+                
+                if (proc == "Test3")
+                {
+                    addr = (nint)(delegate* unmanaged[Cdecl]<byte*, int>)(&WithReturnAndStringMarshal);
+                    return true;
+                }
+                
+                if (proc == "Test4")
+                {
+                    addr = (nint)(delegate* unmanaged[Cdecl]<byte*, byte*, int>)(&WithReturnAndMultiStringMarshal);
+                    return true;
+                }
+                
+                if (proc == "Test5")
+                {
+                    addr = (nint)(delegate* unmanaged[Cdecl]<byte*, int>)(&WithReturnAndSpanMarshal);
+                    return true;
+                }
+                
+                if (proc == "Test6")
+                {
+                    addr = (nint)(delegate* unmanaged[Cdecl]<short, int>)(&WithReturnAndBoolMarshal);
+                    return true;
+                }
+
+                addr = 0;
+                return false;
+            }
+        }
+
+        private TestContext _testContext;
+        private RawAPI _rawAPI;
+
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            _testContext = new TestContext();
+            _rawAPI = new RawAPI(_testContext);
+        }
+        
+        [Benchmark(Baseline = true)]
+        public void RawInvoke()
+        {
+            _rawAPI.RawInvoke();
+        }
+        
+        [Benchmark]
+        public int WithReturn()
+        {
+            return _rawAPI.WithReturn();
+        }
+        
+        [Benchmark]
+        public int WithReturnAndStringMarshal()
+        {
+            return _rawAPI.WithReturnAndStringMarshal("Hello World!");
+        }
+        
+        [Benchmark]
+        public int WithReturnAndMultiStringMarshal()
+        {
+            return _rawAPI.WithReturnAndMultiStringMarshal("Hello ", "World!");
+        }
+
+        [Benchmark]
+        public int WithReturnAndSpanMarshalStackalloc()
+        {
+            return _rawAPI.WithReturnAndSpanMarshal(stackalloc byte[5]);
+        }
+
+        [Benchmark]
+        public int WithReturnAndSpanMarshalLocalBuffer()
+        {
+            Span<byte> buff = new byte[5];
+            return _rawAPI.WithReturnAndSpanMarshal(buff);
+        }
+
+        private static readonly byte[] _buff = new byte[5];
+        [Benchmark]
+        public int WithReturnAndSpanMarshalField()
+        {
+            return _rawAPI.WithReturnAndSpanMarshal(_buff);
+        }
+        
+        [Benchmark]
+        public int WithReturnAndBoolMarshal()
+        {
+            return _rawAPI.WithReturnAndBoolMarshal(true);
+        }
```

Comment:
Future prospect: add a P/Invoke baseline (we can hook into NativeLibrary to map __Internal to the HINSTANCE of the program's module to load UnmanagedCallersOnly pointers)

Reply:
I don't think I understand what you mean? add a P/Invoke override and benchmark that?

Reply:
No I mean do an actual P/Invoke. A virgin `DllImport` with no Silk.NET involved.

Reply:
oh sure, can do that.